### PR TITLE
Remove outdated takari-local-repository extension

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by mosabua — APPROVED** [review]
Provided CI failures are misleading as I expect them to be .. this is good to go.

**Comment by cstamas** [comment]
This extension should be dropped, especially if you use 3.9.x (hopefully 3.9.6) that have this built-in.

To enable file locking you need to specify these: `-Daether.syncContext.named.factory=file-lock` and `-Daether.syncContext.named.nameMapper=file-gav` (or file-hgav that is like file-sha1(gav), so it does not use "deep" paths but git-like paths xx/xx/xxxxx of hashed GAVs).

**Comment by cstamas** [comment]
Almost always `.mvn/maven.config` in form as `-Dfoo=bar` (and keep each param on separate line, Maven 3.9.x enforces this).

The `.mvn/jvm.config` would be needed if "source" of config would be Java Properties. That said, there ARE few Resolver configurations that do come from System Properties (as they are needed "early", before mvn session is created).

The Resolver 2.0.0 doco is much more detailed, so you can see them here (look the "Source" column):
https://maven.apache.org/resolver-archives/resolver-2.0.0-alpha-6/configuration.html

Warning: **this link above is _for Resolver 2.0.0_ and there have been config key renames**, so DO NOT use this page as reference, only to inform yourself which config is sourced from where....

In short: if source is "Session" then it is `-Dfoo=bar` in `.mvn/maven.config`, and it it is "Java System Properties" then it must be set in `.mvn/jvm.config`.

**Comment by wendigo** [comment]
Let's give it a try: https://github.com/trinodb/trino/pull/20467

**Comment by mosabua** [comment]
👋 @electrum @cstamas @wendigo .. do we want to close this since its kinda replaced by some of the other PRs and continue work there?
